code/modules/mob/living/living_health_procs.dm

```diff
@@ -127,20 +127,51 @@
 		S = apply_status_effect(/datum/status_effect/incapacitating/stun, amount)
 	return S
 
+/* DAZE (Light incapacitation) */
+/// Overridable handler to adjust the numerical value of status effects. Expand as needed
+/mob/living/proc/GetDazeDuration(amount)
+	return amount * GLOBAL_STATUS_MULTIPLIER
+/mob/living/proc/IsDaze() //If we're stunned
+	return has_status_effect(/datum/status_effect/incapacitating/dazed)
+/mob/living/proc/AmountDAze() //How many deciseconds remains
+	var/datum/status_effect/incapacitating/dazed/S = IsDaze()
+	if(S)
+		return S.get_duration_left() / GLOBAL_STATUS_MULTIPLIER
+	return 0
 /mob/living/proc/Daze(amount)
-	if(status_flags & CANDAZE)
-		dazed = max(max(dazed,amount),0)
-	return
-
-/mob/living/proc/SetDaze(amount)
-	if(status_flags & CANDAZE)
-		dazed = max(amount,0)
-	return
-
-/mob/living/proc/AdjustDaze(amount)
-	if(status_flags & CANDAZE)
-		dazed = max(dazed + amount,0)
-	return
+	if(!(status_flags & CANDAZE))
+		return
```

Comment:
If we're now using this as a means to set Daze to 0 such as in `/datum/chem_property/positive/neuroshielding/process(mob/living/M, potency = 1, delta_time)` do we want to always perform nothing if they can't be dazed? As in can we guarantee something that has this flag will never be dazed? If not probably should return only if the amount isn't 0.

Reply:
Yeah i've thought about that, i'm not very happy about it either. Debated changing it but that's just how our old code works and i felt it was overarching to change this behavior when it's also used in GODMODE and such. Everything should go through these handlers so it *should* be fine...
/tg/ still has this behavior with signaling and trait cancellation aswell
Can change it though then need to change all status and GODMODE to match

Reply:
I dunno - up to you. Just wanted to bring attention to it.